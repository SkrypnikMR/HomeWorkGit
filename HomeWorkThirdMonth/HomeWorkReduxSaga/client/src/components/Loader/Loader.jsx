import React from 'react';
import './Loader.scss';

const Loader = () => {
    return (
        <div className="loader">
            <div className="loading">
                <p>loading</p>
                <span />
            </div>
        </div>
    );
}

export default Loader;
