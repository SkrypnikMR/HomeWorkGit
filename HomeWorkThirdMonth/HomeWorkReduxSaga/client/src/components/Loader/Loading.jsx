import React from 'react';
import './Loader.scss';

function Loader() {
    return (
        <div className="loader">
            <div className="loading">
                <p>loading</p>
                <span />
            </div>
        </div>
    );
}

export default Loader;
