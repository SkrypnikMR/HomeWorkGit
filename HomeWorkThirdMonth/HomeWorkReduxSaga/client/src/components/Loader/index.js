import Loader from './Loading.jsx';

export default Loader;
