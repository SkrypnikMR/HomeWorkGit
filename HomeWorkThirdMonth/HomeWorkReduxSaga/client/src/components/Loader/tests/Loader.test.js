import React from 'react';
import { shallow, mount } from 'enzyme';
import Loader from '../Loader';

describe('Loader', () => {
    it('Should match snapshot', () => {
        const component = shallow(<Loader />);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should have Loader ', () => {
        const component = mount(<Loader />);
        expect(component.find('Loader')).toHaveLength(1);
    });
    it('Should have divs ', () => {
        const component = mount(<Loader />);
        expect(component.find('div')).toHaveLength(2);
    });
    it('Should have p ', () => {
        const component = mount(<Loader />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('Should have span ', () => {
        const component = mount(<Loader />);
        expect(component.find('span')).toHaveLength(1);
    });
});
