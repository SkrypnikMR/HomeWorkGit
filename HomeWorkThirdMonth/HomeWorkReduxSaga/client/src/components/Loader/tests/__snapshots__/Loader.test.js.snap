// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should match snapshot 1`] = `"<div class=\\"loader\\"><div class=\\"loading\\"><p>loading</p><span></span></div></div>"`;
